### ISSUE TITLE
MobileNetV3 keras models have bug with GlobalAveragePooling2D

### ISSUE THREAD

**Issue by l-bat** [tensorflow/tensorflow#48504]
**System information**
TensorFlow version:
```python
print(tf.version.GIT_VERSION, tf.version.VERSION)
v2.4.0-rc4-71-g582c8d236cb 2.4.0
```

**Describe the current behavior**
According to [MobileNetV3Small documentation](https://www.tensorflow.org/api_docs/python/tf/keras/applications/MobileNetV3Small)

> The weights for all 6 models are obtained and translated from the Tensorflow checkpoints from TensorFlow checkpoints found here.

But keras models cannot reproduce performance on ImageNet #48066.

Tensorflow pb model contains GlobalAvgPool **before** Conv2D with filter <1x1x576x1024>:
![image](https://user-images.githubusercontent.com/22346860/114558652-93379e80-9c73-11eb-8165-5d5230c0494c.png)

But MobileNetV3Small from keras.application contains GlobalAvgPool **after** Conv2D with filter <1x1x576x1024>:
![image](https://user-images.githubusercontent.com/22346860/114558795-b82c1180-9c73-11eb-95e4-3f1eb05dc374.png)

https://github.com/tensorflow/tensorflow/blob/6a278b9cf2fbed781661079b13d88b4106622a09/tensorflow/python/keras/applications/mobilenet_v3.py#L295-L308

Such version of MobileNetV3Small achieves top1: 56.79% instead of 68.1%.

**Describe the expected behavior**
To achieve reported accuracy, change [model code](https://github.com/tensorflow/tensorflow/blob/6a278b9cf2fbed781661079b13d88b4106622a09/tensorflow/python/keras/applications/mobilenet_v3.py#L295-L315) to:

```python
  x = activation(x)

  x = layers.GlobalAveragePooling2D()(x)
  if channel_axis == 1:
    x = layers.Reshape((last_conv_ch, 1, 1))(x)
  else:
    x = layers.Reshape((1, 1, last_conv_ch))(x)

  x = layers.Conv2D(
      last_point_ch,
      kernel_size=1,
      padding='same',
      use_bias=True,
      name='Conv_2')(x)
  x = activation(x)

  if include_top:
    if dropout_rate > 0:
      x = layers.Dropout(dropout_rate)(x)
    x = layers.Conv2D(classes, kernel_size=1, padding='same', name='Logits')(x)
    x = layers.Flatten()(x)
    imagenet_utils.validate_activation(classifier_activation, weights)
    x = layers.Activation(activation=classifier_activation,
                          name='Predictions')(x)
```

**Comment by fchollet** [comment]
Thanks for the report. Please open a PR with the fix.

**Comment by qlzh727** [comment]
Thanks for reporting the issue. Please https://github.com/tensorflow/tensorflow/pull/48542#issuecomment-824258134 for more updates. #48542 should fix the issue.
